python/ccf/ledger.py

```diff
@@ -513,6 +518,68 @@ def last_verified_txid(self):
             self._ledger_validator.last_verified_seqno,
         )
 
+    def get_transaction(self, seqno: int) -> Transaction:
+        """
+        Returns the :py:class:`ccf.Ledger.Transaction` recorded in the ledger at the given sequence number.
+
+        Note that the transaction returned may not yet be verified by a
+        signature transaction nor committed by the service.
+
+        :param int seqno: Sequence number of the transaction to fetch.
+
+        :return: :py:class:`ccf.Ledger.Transaction`
+        """
+        if seqno < 1:
+            raise ValueError("Ledger first seqno is 1")
+
+        self._reset_iterators()
+
+        transaction = None
+        try:
+            for chunk in self:
+                for tx in chunk:
```

Comment:
Non-critical: This iteration is going to be very slow. Could we rely on the filenames here to scan to the relevant chunk, and parse it in isolation? And even within that, could we skip over other transactions rather than parsing them? Not needed for this PR or current tests, but will likely bite us if we start using this `ledger` util in more places/on real service ledgers.

Reply:
Yep, I should have mentioned it in the comment. I wanted the fetching to still verify the integrity of the ledger and we currently have no way to do that without starting from the very beginning of the ledger. I'll create a separate ticket and add a comment in the docstring.

Reply:
@eddyashton See first item in https://github.com/microsoft/CCF/issues/2453 for follow-up improvements.